portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -57,27 +55,30 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         PreparedQuery results = datastore.prepare(query);
  
         List<String> pastComments = new ArrayList<>();
-        int validNumRequest = 0;
-        int numCommentsWanted = 0;
-        boolean numCommentsWantedTrigger = true;
  
         for (Entity entity : results.asIterable()) {
-            validNumRequest = Integer.parseInt((String)(entity.getProperty("seeNumComments")));
-            if (validNumRequest >= 0){
+            // validNumInput specifies if the form input for number of comments wanted is valid
+                // (meaning >= 0). If not, then the comment entered in that entity won't be shown on the site.
+            int validNumInput = 0;
+            int numCommentsWanted = 0;
+            boolean setNumCommentsWanted = true;            
+            validNumInput = Integer.parseInt((String)(entity.getProperty("seeNumComments")));
+            
+            if (validNumInput >= 0) {
                 String comment = (String) (entity.getProperty("newComment"));
 
-                //For testing queried result comments
-                // System.out.println("Query result comment: " + comment);
+                // For testing queried result comments
+                //System.out.println("Query result comment: " + comment);
 
                 pastComments.add(comment);
             } else {
-                System.out.println("Input of number comments wanted not greater than or equal to 0: " + validNumRequest);
+                System.out.println("Number of comments wanted is not greater than or equal to 0: " + validNumInput);
             }
            
-            //Grab number of comments wanted to be seen by user only in the first entity
-            while (numCommentsWantedTrigger) {
-                numCommentsWanted = validNumRequest;
-                numCommentsWantedTrigger = false;
+            // Grab number of comments wanted to be seen by user only in the first entity
+            if (numCommentsWantedTrigger) {
```

Comment:
I could be wrong here, but the way I read this logic is that if the first comment doesn't have a valid input num, you'll still hit this block of code and set numCommentsWanted = 0.  If you want this to only trigger after you've gotten a valid number of comments, you'll probably want to move it up into the block above where you've checked validNumInput >= 0.

Reply:
Fixed by doing valid input checking in doPost rather than in doGet so as to not put bad data in the Datastore